sql/src/main/java/io/crate/planner/operators/Collect.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner.operators;
+
+import io.crate.action.sql.SessionContext;
+import io.crate.analyze.OrderBy;
+import io.crate.analyze.QueriedTableRelation;
+import io.crate.analyze.WhereClause;
+import io.crate.analyze.symbol.Symbol;
+import io.crate.metadata.table.TableInfo;
+import io.crate.planner.Plan;
+import io.crate.planner.Planner;
+import io.crate.planner.PositionalOrderBy;
+import io.crate.planner.distribution.DistributionInfo;
+import io.crate.planner.node.dql.RoutedCollectPhase;
+import io.crate.planner.projection.builder.ProjectionBuilder;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+class Collect implements LogicalPlan {
+
+    final QueriedTableRelation relation;
+    final WhereClause where;
+
+    final List<Symbol> toCollect;
+    final TableInfo tableInfo;
+
+    Collect(QueriedTableRelation relation, List<Symbol> toCollect, WhereClause where, Set<Symbol> usedColumns) {
+        this.relation = relation;
+        this.where = where;
+        this.tableInfo = relation.tableRelation().tableInfo();
+        this.toCollect = toCollect;
+    }
+
+    @Override
+    public Plan build(Planner.Context plannerContext,
+                      ProjectionBuilder projectionBuilder,
+                      int limit,
+                      int offset,
+                      @Nullable OrderBy order) {
+        SessionContext sessionContext = plannerContext.transactionContext().sessionContext();
+        RoutedCollectPhase collectPhase = new RoutedCollectPhase(
+            plannerContext.jobId(),
+            plannerContext.nextExecutionPhaseId(),
+            "collect",
```

Comment:
It's not really clear to me what the "collect" string is meant to represent (a command, or id, or something else?) 
Maybe it'd help to have this as a constant and give it a meaningful name? (COLLECT_PHASE_NAME maybe? or something similar?)  
There are several places in this PR (basically in all LogicalPlans)

Reply:
the name of the phases is exposed via `sys.operations` - I kept the old name and would like to keep it as I don't want to change any behaviour with the refactoring. I can add a constant.

Not sure what you're referring to with "basically in all LogicalPlans". The only other LogicalPlan that has something like this is Count?